flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/utils/RexNodeExtractor.scala

```diff
@@ -532,7 +531,14 @@ class RexNodeToExpressionConverter(
     Try(functionCatalog.lookupFunction(identifier)) match {
       case Success(f: java.util.Optional[FunctionLookup.Result]) =>
         if (f.isPresent) {
-          Some(new CallExpression(f.get().getFunctionDefinition, operands, outputType))
+          val identifier = f.get.getFunctionDefinition match {
```

Comment:
we could simplify this logic if FLINK-23384 would be fixed, I will add a comment above those lines